File: app/src/main/java/com/elitec/alejotaller/feature/agent/domain/caseUse/CheckAgentConnectionCaseUse.kt
package com.elitec.alejotaller.feature.agent.domain.caseUse

import com.elitec.alejotaller.feature.agent.domain.entity.AgentConnectionResult
import com.elitec.alejotaller.feature.agent.domain.repository.AgentRepository

class CheckAgentConnectionCaseUse(
    private val repository: AgentRepository,
) {
    suspend operator fun invoke(): AgentConnectionResult =
        repository.checkConnection()
}

File: web/src/core/feature/agent/domain/caseuse/CheckAgentConnectionCaseUse.ts
import type { AgentConnectionResult } from "../entity/AgentMessage";
import type { AgentRepository } from "../repository/agent.repository";

export class CheckAgentConnectionCaseUse {
  constructor(private readonly repo: AgentRepository) {}

  execute(): Promise<AgentConnectionResult> {
    return this.repo.checkConnection();
  }
}
